forwarder/writer_test.go

```diff
@@ -870,6 +871,9 @@ func TestForwarderWrite(t *testing.T) {
 		// Not using any of the returned values currently
 		_, _ = f.Write(context.Background(), test.arg)
 
+		// give one second for mock go routines to complete
+		time.Sleep(1500 * time.Millisecond)
```

Comment:
I don't like seeing `time` used in tests. This slows down the whole test suite. If `fwd.Stop()` blocks until all workers are done writing their work, then you could call `f.Write(...)` then `ff.Stop()` and then be ready to run through all the test cases.